src/main/java/com/openlattice/datastore/data/controllers/DataController.java

```diff
@@ -414,6 +417,51 @@ public Void clearEntitySet( UUID entitySetId ) {
         return null;
     }
 
+    @Override
+    @RequestMapping(
+            path = { "/" + ENTITY_SET + "/" + SET_ID_PATH + "/" + ENTITY_KEY_ID_PATH + "/graph" },
+            method = RequestMethod.DELETE
+    )
+    public Void clearEntityAndNeighborEntities(
+            @PathVariable( ENTITY_SET_ID ) UUID entitySetId,
+            @PathVariable( ENTITY_ID ) UUID entityId
+    ) {
+
+        ensureReadAccess( new AclKey( entitySetId ) );
+        Map<UUID, List<NeighborEntityDetails>> result = searchService
+                .executeEntityNeighborSearch( entitySetId, ImmutableSet.of( entityId ) );
+
+        if ( result != null && !result.containsKey( entityId ) ) {
+
+            List<NeighborEntityDetails> neighbors = result.get( entityId );
+            neighbors.parallelStream().forEach( neighbor -> {
+
+                UUID associationEntityId = (UUID) neighbor.getAssociationDetails().get( ID_FQN ).iterator().next();
+                UUID associationEntitySetId = neighbor.getAssociationEntitySet().getId();
+                Map<UUID, PropertyType> authorizedPropertyTypesOnNeighborAES = authzHelper
```

Comment:
Theoretically you can have write permissions on entity set properties without having write on the entity set itself -- unclear what we should do in this case. Do we need to also check for write permissions on the entity set? (Also not necessarily related to this PR, but what is expected behavior if someone calls delete on an entity and only has write permissions on some of the properties?)

Reply:
Not sure what the behavior should be; definitely worth discussing in longer form.